### DISCUSSION TITLE
What does the info in test duration mean?

### DISCUSSION THREAD

**Discussion by kingyue737** [vitest-dev/vitest#1770] [Q&A] [answered]
In the latest vitest, more info is shown in test duration.
In the following case `setup` is longer than total time, and sometimes I found that `collect` can also longer than `time`. What do they mean?

![image](https://user-images.githubusercontent.com/40021217/182290134-8cdb992e-4a30-481b-ac64-f1ae3b622c49.png)

**Comment by sheremet-va** [comment, marked as answer]
`setup` is a total sum of `setupFiles` for each test file, `collect` is importing your test files (includes transformations), `tests` is actually running your code.

Your overall time might be smaller because we run tests in parallel, but we show actual time that it took to do something.

**Reply by DevathiMahesh** [reply]
@sheremet-va  I have a similar problem. In my case transform is taking 1.8 sec, tests taking 2 sec, setup taking 13 seconds but collect is taking 158 sec. I have a swc setup in vite & disabled esbuild, bcoz of that transformations are very fast. It took only 1.8 sec. Is there any way to reduce the **collect** time so that my overall running time will be reduced??? If you have any idea, please help me out in this regard......

**Reply by douglasg14b** [reply]
Same thing here, `collect` takes the longest, how do we improve that?

**Comment by LeoAnesi** [comment]
There are new attributes now which are `transform`, `environment` and `prepare`, could you explain what they mean too @sheremet-va ?

**Reply by sheremet-va** [reply]
- transform: how long transform phase take (how long vite transformed these files)
- environment: how long did environment load (e.g., importing jdom + updating globals)
- prepare: how long it took Vitest to import and execute code before actually running tests
